js/src/forum/utils/reply.js

```diff
@@ -1,9 +1,10 @@
 import DiscussionControls from 'flarum/utils/DiscussionControls';
 import EditPostComposer from 'flarum/components/EditPostComposer';
+import cleanDisplayName from './cleanDisplayName';
 
 function insertMention(post, composer, quote) {
   const user = post.user();
-  const mention = '@' + (user ? user.username() : post.number()) + '#' + post.id() + ' ';
+  const mention = `@"${(user && cleanDisplayName(user)) || app.translator.trans('core.lib.username.deleted_text')}"#p${post.id()}`;
```

Comment:
@SychO9 Found an issue while working on fixing up the editor. This needs to include a space after the mention.

Reply:
oh yeah, I removed the space at the end 🙈 I just pushed a commit to master that fixes it.

Reply:
Great, thanks!